cli/validation/converter/iam_test.go

```diff
@@ -0,0 +1,330 @@
+				Type: "cloudresourcemanager.googleapis.com/Organization",
+				IAMPolicy: &cai.IAMPolicy{
+					Bindings: []cai.IAMBinding{
+						{
+							Role:    "testRole1",
+							Members: []string{"member1"},
+						},
+						{
+							Role:    "testRole2",
+							Members: []string{"member1", "member2"},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "folder iam binding",
+			inputs: []cai.Resource{
+				{
+					Name:    "resource",
+					Type:    "gcp-types/cloudresourcemanager-v2:virtual.folders.iamMemberBinding",
+					Project: "project",
+					Properties: map[string]interface{}{
+						"resource": "folders/123456",
+						"role":     "testRole",
+						"member":   "member",
+					},
+				},
+			},
+			output: cai.Asset{
+				Name: "//cloudresourcemanager.googleapis.com/folders/123456",
+				Type: "cloudresourcemanager.googleapis.com/Folder",
+				IAMPolicy: &cai.IAMPolicy{
+					Bindings: []cai.IAMBinding{
+						{
+							Role:    "testRole",
+							Members: []string{"member"},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "project iam binding",
+			inputs: []cai.Resource{
+				{
+					Name:    "resource",
+					Type:    "gcp-types/cloudresourcemanager-v1:virtual.projects.iamMemberBinding",
+					Project: "project",
+					Properties: map[string]interface{}{
+						"resource": "123456",
+						"role":     "testRole",
+						"member":   "member2",
+					},
+				},
+				{
+					Name:    "resource",
+					Type:    "gcp-types/cloudresourcemanager-v1:virtual.projects.iamMemberBinding",
+					Project: "project",
+					Properties: map[string]interface{}{
+						"resource": "123456",
+						"role":     "testRole",
+						"member":   "member1",
+					},
+				},
+				{
+					Name:    "resource",
+					Type:    "gcp-types/cloudresourcemanager-v1:virtual.projects.iamMemberBinding",
+					Project: "project",
+					Properties: map[string]interface{}{
+						"resource": "123456",
+						"role":     "testRole",
+						"member":   "member3",
+					},
+				},
+			},
+			output: cai.Asset{
+				Name: "//cloudresourcemanager.googleapis.com/projects/123456",
+				Type: "cloudresourcemanager.googleapis.com/Project",
+				IAMPolicy: &cai.IAMPolicy{
+					Bindings: []cai.IAMBinding{
+						{
+							Role:    "testRole",
+							Members: []string{"member1", "member2", "member3"},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			var asset cai.Asset
+			for _, res := range c.inputs {
+				newAsset, err := GetIamCaiObject(res.Type, res)
+				if err != nil {
+					t.Errorf("got error: %t", err)
+				}
+				if asset.Type != "" {
+					asset, err = MapIamCaiObject(asset, newAsset)
+					if err != nil {
+						t.Errorf("got error: %t", err)
+					}
+				} else {
+					asset = newAsset
+				}
+			}
+
+			if !reflect.DeepEqual(asset, c.output) {
+				t.Errorf("got %v, expected %v", asset.IAMPolicy, c.output.IAMPolicy)
+			}
+		})
+	}
+}
+
+func TestMergeBindings(t *testing.T) {
```

Comment:
Any reason to duplicate this? https://github.com/GoogleCloudPlatform/terraform-google-conversion/blob/master/google/iam_helpers_test.go#L8

If we're copying the logic, we should probably copy the cases.

Reply:
I've added cases that haven't been covered by the conversion library.
cc @sjvanrossum

Reply:
Then just add those cases on top of the existing ones. There's no reason to change the structure/logic.

In general, that's my biggest feedback: I don't want to see code drift when we're doing essentially the same thing.

Reply:
It makes sense to me to add new tests to cover different function and set of logic. That's how unit tests work.

Do you want all unit tests to be in a single test case?